infra/Pos/Slotting/Impl/Simple.hs

```diff
@@ -42,49 +51,49 @@ data SimpleSlottingState = SimpleSlottingState
     { _sssLastSlot :: !SlotId
     }
 
-type SimpleSlottingVar = TVar SimpleSlottingState
+type SimpleSlottingStateVar = TVar SimpleSlottingState
 
-mkSimpleSlottingVar :: (MonadIO m, HasConfiguration) => m SimpleSlottingVar
-mkSimpleSlottingVar = atomically $ newTVar $ SimpleSlottingState $ unflattenSlotId 0
+mkSimpleSlottingStateVar :: (MonadIO m, HasConfiguration) => m SimpleSlottingStateVar
+mkSimpleSlottingStateVar = atomically $ newTVar $ SimpleSlottingState $ unflattenSlotId 0
 
 ----------------------------------------------------------------------------
 -- Implementation
 ----------------------------------------------------------------------------
 
 getCurrentSlotSimple
-    :: (SimpleSlottingMode ctx m)
-    => SimpleSlottingVar
-    -> m (Maybe SlotId)
-getCurrentSlotSimple var = traverse (updateLastSlot var) =<< (currentTimeSlottingSimple >>= slotFromTimestamp)
+    :: (MonadSimpleSlotting ctx m)
+    => m (Maybe SlotId)
+getCurrentSlotSimple = do
+    var <- view (lensOf @SimpleSlottingStateVar)
+    traverse (updateLastSlot var) =<< (currentTimeSlottingSimple >>= slotFromTimestamp)
 
 getCurrentSlotBlockingSimple
-    :: (SimpleSlottingMode ctx m)
-    => SimpleSlottingVar
-    -> m SlotId
-getCurrentSlotBlockingSimple var = do
+    :: (MonadSimpleSlotting ctx m)
+    => m SlotId
+getCurrentSlotBlockingSimple = do
     (_, nextEpochIndex) <- getCurrentNextEpochIndexM
-    getCurrentSlotSimple var >>= \case
+    getCurrentSlotSimple >>= \case
         Just slot -> pure slot
         Nothing -> do
             waitCurrentEpochEqualsM nextEpochIndex
-            getCurrentSlotBlockingSimple var
+            getCurrentSlotBlockingSimple
 
 getCurrentSlotInaccurateSimple
-    :: (SimpleSlottingMode ctx m)
-    => SimpleSlottingVar
-    -> m SlotId
-getCurrentSlotInaccurateSimple var =
-    getCurrentSlotSimple var >>= \case
+    :: (MonadSimpleSlotting ctx m)
+    => m SlotId
+getCurrentSlotInaccurateSimple =
```

Comment:
These previous 3 functions have all changed to take their slotting var from a typeclass rather than as a parameter. Any specific advantage to this?

Reply:
That's true, but it now plays the role of `getCurrentSlotSum` where the argument was read from a typeclass.  I haven't figure out if we can change this. But since we want to remove `MonadSlots` this requirement should go away as well.